### PR TITLE
Interface: conformance test of Interface

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
part of #13653

CHANGELOG_BEGIN
CHANEGLO_END

### CODE REVIEW

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/InterfaceIT.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.TransactionHelpers._
+import com.daml.ledger.api.v1.value.Identifier
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.semantic.Interface._
+import scalaz.Tag
+
+class InterfaceIT extends LedgerTestSuite {
```
Comment (remyhaemmerle-da) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/InterfaceIT.scala:18]: @sofiafaro-da Do you have idea what we should test here ?

Comment (remyhaemmerle-da) [reply ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/InterfaceIT.scala:21]: The current implementation does not offer type checking feature. 
It is a separate bullet #13653

**Comment by remyhaemmerle-da** [comment]
@sofiafaro-da Are you fine with merging this PR as and add your suggesting once we extended the API.

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!

**Comment by remyhaemmerle-da** [comment]
@rgugliel-da @phoebe-da This is the PR with conformance test for interface. 
I enabled it for canton.

**Review by cocreature — APPROVED** [review]
Bazel changes look good :+1:

**Comment by phoebe-da** [comment]
> @rgugliel-da @phoebe-da This is the PR with conformance test for interface.
I enabled it for canton.

Is it running against Canton in your CI? (And if so, why isn't it failing?)

**Comment by phoebe-da** [comment]
> @rgugliel-da @phoebe-da This is the PR with conformance test for interface.
I enabled it for canton.

Is it running against Canton in your CI? (And if so, why isn't it failing?)

**Comment by remyhaemmerle-da** [comment]
@phoebe-da wrote:
> Is it running against Canton in your CI? (And if so, why isn't it failing?)

We updated canton this morning (See #13905), I was under the impression it should success with this version, shouldn't it ?

**Comment by phoebe-da** [comment]
> I was under the impression it should success with this version, shouldn't it ?

It depends what the test is doing, but I think no. Canton is using lenient exercise at the moment, and the change to get rid of that is only in a branch.

**Comment by phoebe-da** [comment]
> I was under the impression it should success with this version, shouldn't it ?

It depends what the test is doing, but I think no. Canton is using lenient exercise at the moment, and the change to get rid of that is only in a branch.

**Comment by remyhaemmerle-da** [comment]
> It depends what the test is doing, but I think no.  Canton is using lenient exercise at the moment, and the change to get rid of that is only in a branch.

The test does not tackle clashing name, so it should work with Lenient exercise.

**Comment by phoebe-da** [comment]
Makes sense -- is there a way to test the clashing name case atm?

**Comment by sofiafaro-da** [comment]
> Makes sense -- is there a way to test the clashing name case atm?

Right now we still have the name collision check in place (including in the engine) that prevents these choice name clashes in LF, so it's not possible to test atm. Once we relax that we can add a test.

**Pull request merged**